python/hsfs/constructor/fs_query.py

```diff
@@ -99,16 +99,21 @@ def register_external(
             Union[TypeVar("pyspark.sql.DataFrame"), TypeVar("pyspark.RDD")]
         ] = None,
     ) -> None:
-        if self._on_demand_fg_aliases is None:
-            return
-
-        for external_fg_alias in self._on_demand_fg_aliases:
-            if type(external_fg_alias.on_demand_feature_group).__name__ == "SpineGroup":
-                external_fg_alias.on_demand_feature_group.dataframe = spine
-            engine.get_instance().register_external_temporary_table(
-                external_fg_alias.on_demand_feature_group,
-                external_fg_alias.alias,
-            )
+        if self._on_demand_fg_aliases is not None:
+            for external_fg_alias in self._on_demand_fg_aliases:
+                if type(external_fg_alias.on_demand_feature_group).__name__ == "SpineGroup":
+                    external_fg_alias.on_demand_feature_group.dataframe = spine
+                engine.get_instance().register_external_temporary_table(
+                    external_fg_alias.on_demand_feature_group,
+                    external_fg_alias.alias,
+                )
+
+        if self._hudi_cached_feature_groups is not None:
```

Comment:
why do you need to register hudi temp table here? is it not registered by `register_hudi_tables`?